captum/optim/models/_common.py

```diff
@@ -1,322 +1,229 @@
+                new_layer = _transfer_layer_vars(child, layer2, **kwargs)
+            else:
+                new_layer = layer2(**kwargs)
+            setattr(model, name, new_layer)
+        elif child is not None:
+            replace_layers(child, layer1, layer2, transfer_vars, **kwargs)
+
+
+def _transfer_layer_vars(
+    layer1: nn.Module, layer2: Type[nn.Module], **kwargs
+) -> nn.Module:
+    """
+    Given a layer instance, create a new layer instance of another class
+    with the same initialization variables as the original layer.
+    Args:
+        layer1: (nn.Module): A layer instance that you want to transfer
+            initialization variables from.
+        layer2: (nn.Module): The layer class to create with the variables
+            from of layer1.
+        kwargs: (Any, optional): Any additional variables to use when creating
+            the new layer.
+    Returns:
+        layer2 instance (nn.Module): An instance of layer2 with the initialization
+            variables that it shares with layer1, and any specified additional
+            initialization variables.
+    """
+
+    l2_vars = list(signature(layer2.__init__).parameters.values())
+    l2_vars = [
+        str(l2_vars[i]).split()[0]
+        for i in range(len(l2_vars))
+        if str(l2_vars[i]) != "self"
+    ]
+    l2_vars = [p.split(":")[0] if ":" in p and "=" not in p else p for p in l2_vars]
+    l2_vars = [p.split("=")[0] if "=" in p and ":" not in p else p for p in l2_vars]
+    layer2_vars: Dict = {k: [] for k in dict.fromkeys(l2_vars).keys()}
+
+    layer1_vars = {k: v for k, v in vars(layer1).items() if not k.startswith("_")}
+    shared_vars = {k: v for k, v in layer1_vars.items() if k in layer2_vars}
+    new_vars = dict(item for d in (shared_vars, kwargs) for item in d.items())
+    return layer2(**new_vars)
+
+
+class Conv2dSame(nn.Conv2d):
+    """
+    Tensorflow like 'SAME' convolution wrapper for 2D convolutions.
+    TODO: Replace with torch.nn.Conv2d when support for padding='same'
+    is in stable version
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Tuple[int, int]],
+        stride: Union[int, Tuple[int, int]] = 1,
+        padding: Union[int, Tuple[int, int]] = 0,
+        dilation: Union[int, Tuple[int, int]] = 1,
+        groups: int = 1,
+        bias: bool = True,
+    ) -> None:
+        super(Conv2dSame, self).__init__(
+            in_channels, out_channels, kernel_size, stride, 0, dilation, groups, bias
+        )
+
+    def calc_same_pad(self, i: int, k: int, s: int, d: int) -> int:
+        return max((math.ceil(i / s) - 1) * s + (k - 1) * d + 1 - i, 0)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        ih, iw = x.size()[-2:]
+        kh, kw = self.weight.size()[-2:]
+        pad_h = self.calc_same_pad(ih, kh, self.stride[0], self.dilation[0])
+        pad_w = self.calc_same_pad(iw, kw, self.stride[1], self.dilation[1])
```

Comment:
TorchScript / JIT seems to throw an error about `k` not having an assigned value and it can be resolved by simply setting what each variable is for like so:

```
        pad_h = self.calc_same_pad(i=ih, k=kh, s=self.stride[0], d=self.dilation[0])
        pad_w = self.calc_same_pad(i=iw, k=kw, s=self.stride[1], d=self.dilation[1])
```